app/src/test/java/com/example/main_animal_villa/CharacterTests.kt

```diff
@@ -0,0 +1,108 @@
+package com.example.main_animal_villa
+
+import androidx.arch.core.executor.testing.InstantTaskExecutorRule
+import androidx.lifecycle.Observer
+import com.animal_villa.MainViewModel
+import com.animal_villa.dto.Character
+import io.mockk.MockKAnnotations
+import io.mockk.coEvery
+import io.mockk.impl.annotations.MockK
+import junit.framework.Assert.assertNotNull
+import junit.framework.Assert.assertTrue
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.newSingleThreadContext
+import kotlinx.coroutines.test.resetMain
+import kotlinx.coroutines.test.runTest
+import kotlinx.coroutines.test.setMain
+import org.junit.*
+import org.junit.rules.TestRule
+import java.util.*
+import java.util.concurrent.CountDownLatch
+import java.util.concurrent.TimeUnit
+
+class CharacterTests {
+
+    @get:Rule
+    var rule : TestRule = InstantTaskExecutorRule()
+    var character : Character? = Character()
+
+    lateinit var characterService : CharacterService
+
+    lateinit var mvm : MainViewModel
```

Comment:
late initialization lets you declare these variables without having to instantiate them. This will let you test different scenarios while letting you declare the variable at a higher scope. More information on lateinit here: https://kotlinlang.org/docs/properties.html#late-initialized-properties-and-variables